src/pypromice/process/join_l3.py

```diff
@@ -0,0 +1,282 @@
+    #         - df_aux.loc[df_aux.z_surf_combined.last_valid_index(), 'z_surf_combined'] \
+    #          + df_v6.loc[df_v6.z_surf_combined.first_valid_index(), 'z_surf_combined'] 
+    
+    # if s == 'Swiss Camp 10m':        
+    #     df.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] = \
+    #         df2.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] \
+    #             - df2.loc[df2.HS_combined.last_valid_index(), 'HS_combined'] \
+    #              + df.loc[df.HS_combined.first_valid_index(), 'HS_combined'] 
+
+                    
+    # df.loc[df.HS_combined.diff()==0,'HS_combined'] = np.nan
+    
+    # fit = np.polyfit(df.loc[df.HS_combined.notnull(),:].index.astype('int64'),  
+    #                    df.loc[df.HS_combined.notnull(),'HS_combined'], 1)
+    # fit_fn = np.poly1d(fit)
+    
+    # df['HS_combined'] = df['HS_combined'].values \
+    #     - fit_fn(
+    #         df_in.loc[[df_in.z_surf_combined.first_valid_index()],:].index.astype('int64')[0]
+    #     )  +  df_in.loc[df_in.z_surf_combined.first_valid_index(), 'z_surf_combined']
+    # return l3m
+    
+def build_station_dict(config_folder):
+    station_dict = {}
+    
+    # Iterate through all files in the given folder
+    for filename in os.listdir(config_folder):
+        if filename.endswith(".toml"):
+            # Construct full file path
+            file_path = os.path.join(config_folder, filename)
+            
+            # Load TOML file
+            with open(file_path, 'r') as file:
+                data = toml.load(file)
+                station_key = next(iter(data))
+                station_data = data[station_key]
+               
+                if station_data["station_site"] in station_dict:
+                    station_dict[station_data["station_site"]].append(station_key)
+                else:
+                    station_dict[station_data["station_site"]] = [station_key]
+    
+    return station_dict
+    
+def join_l3():
+    args = parse_arguments_joinl3()
+    logging.basicConfig(
+        format="%(asctime)s; %(levelname)s; %(name)s; %(message)s",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    
+    station_dict = build_station_dict(args.config_folder)
+
+    l3m = xr.Dataset()
+    for stid in  station_dict[args.site]:
+        logger.info(stid)
+        
+        is_promice = False
+        is_gcnet = False
+        filepath = os.path.join(args.folder_l3, stid, stid+'_hour.nc')
+        if os.path.isfile(filepath):
+            is_promice = True
+        else:
+            filepath = os.path.join(args.folder_gcnet, stid+'.csv')
+            if os.path.isfile(filepath):
+                is_gcnet = True
+        if not is_promice and not is_gcnet:            
+            logger.info(stid+' not found either in '+args.folder_l3+' or '+args.folder_gcnet)
+            continue
+
+        l3, _ = loadArr(filepath)
+        
+        if is_gcnet:
+            l3 = gcnet_postprocessing(l3)
```

Comment:
I would have expected the historical data files to have been converted to a compatible L3 format prior this script.